content/400-reference/200-api-reference/050-prisma-client-reference.mdx

```diff
@@ -5429,7 +5429,17 @@ const getUsers = await prisma.user.findMany({
 
 ## Client methods
 
-Client-level methods are prefixed by `$`.
+### Remarks
+
+- Client-level methods are prefixed by `$`.
+- `$on` and `$use` do not exist on clients extended using `$extends`
+
+<Admonition type="warning">
+  In extended clients, Client methods do not necessarily exist. If you are extending your client, make sure to check for existence before using Client methods like `$transaction` or `$connect`.
+
+  In addition, if you are using `$on` or `$use` in extended clients you can either transition [to use query extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/query) or hook up changes prior to extending your client.
```

Comment:
How is `$on` related to query extension component type? Isn't it just middlewares (so `$use`)?

Reply:
you're right. clarified this language.